src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/integration/MessagingMultiNodeTestBase.kt

```diff
@@ -68,7 +68,11 @@ abstract class MessagingMultiNodeTestBase : IntegrationTestBase() {
       val allEvents = telemetryRepository.findAllByEvent(event.eventName)
       val matchingEvents = allEvents?.filter {
         expected.entries.map { (k, v) ->
-          JSONObject(it.properties).get(k).equals(v)
+          val jsonObject = JSONObject(it.properties)
+          when {
+            (jsonObject.has(k)) -> jsonObject.get(k).equals(v)
+            else -> false
+          }
```

Comment:
What's this for?

Reply:
This fixes an issue if you have 2 separate events with different maps. If you search for a property that don't exist on the event it throws a JSON exception. This way it checks all regardless of property map